src/query/changed.go

```diff
@@ -0,0 +1,94 @@
+package query
+
+import (
+	"core"
+	"scm"
+)
+
+// ChangedRequest is a simple parameter object
+type ChangedRequest struct {
+	Since            string
+	DiffSpec         string
+	IncludeDependees string
+}
+
+// ChangedLabels returns all BuildLabels that have changed according to the given request.
+func ChangedLabels(state *core.BuildState, request ChangedRequest) []core.BuildLabel {
+	workspaceChangedFiles := changedFiles(request.Since, request.DiffSpec)
+	if len(workspaceChangedFiles) == 0 {
+		return []core.BuildLabel{}
+	}
+
+	labels := make([]core.BuildLabel, 0)
```

Comment:
nit: I probably wouldn't bother with `make` if it's got an initial capacity of 0, just `[]core.BuildLabel{}` is a little simpler